src/main/java/com/ericsson/ei/subscription/SubscriptionRepeatDbHandler.java

```diff
@@ -64,13 +66,12 @@
      * @param aggrObjId
      */
     public void addMatchedAggrObjToSubscriptionId(String subscriptionId,
-            int requirementId, String aggrObjId) {
+            int requirementId, String aggrObjId, Map<String, List<String>> subscriptionsCache) {
         LOGGER.debug(
                 "Adding/Updating matched AggrObjId: {} to SubscriptionsId: {} aggrId matched list",
                 aggrObjId, subscriptionId);
 
-        if (checkIfAggrObjIdExistInSubscriptionAggrIdsMatchedList(subscriptionId, requirementId,
-                aggrObjId)) {
+        if (checkIfAggrObjIdExistInSubscriptionAggrIdsMatchedList(subscriptionId, requirementId, aggrObjId, true, subscriptionsCache)) {
```

Comment:
Here we should not use the cache, but maybe we can skip this check as is already done before calling this